### PR TITLE
feat: add pre commit hook for linting

### LINKED ISSUE #1030: Add a pre-commit hook for linting files
**Desired Behavior**
Lint files using a pre-commit hook 

---
[Optional] **Alternatives**

[Optional] **Additional context**

---
**Contributor Resources**
The `layer5` repo contains two websites. The current generation and the next-generation of the layer5.io site.

If the `layer5-ng` label is absent on this issue, then this issue pertains to the current generation of the layer5.io website, which uses Jekyll and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- See [`layer5` contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md)

If the `layer5-ng` label is present on this issue, then this issue pertains to the next-generation of the layer5.io website, which uses Gatsby, Strapi, and GitHub Pages. Site content is found under the [`layer5-ng` branch](https://github.com/layer5io/layer5/tree/layer5-ng).
- See [`layer5-ng` contributing instructions](https://github.com/layer5io/layer5/blob/layer5-ng/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by Tanuj22**
**Description**

This PR fixes #1030

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -71,7 +70,18 @@
     "cpx": "^1.5.0",
     "cross-env": "^7.0.0",
     "env-cmd": "^10.1.0",
-    "prettier": "^1.19.1",
+    "husky": "^4.3.0",
+    "lint-staged": "^10.5.0",
```
Comment (kushthedude) [inline package.json:74]: What is this dep used for?

File: package.json [hunk 2]
```diff
@@ -71,7 +70,18 @@
...
     "cross-env": "^7.0.0",
     "env-cmd": "^10.1.0",
-    "prettier": "^1.19.1",
+    "husky": "^4.3.0",
+    "lint-staged": "^10.5.0",
     "rimraf": "^3.0.2"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
```
Comment (kushthedude) [inline package.json:79]: Why can't we use `npm run lint` over here?

Comment (Tanuj22) [reply package.json:79]: Using `lint-staged` will only run eslint on files to be committed. Directly using `npm run lint` will run it on all files making it slow.

Comment (kushthedude) [reply package.json:79]: Oh, that's good! Can you also do this in Meshery 😄 ?

Comment (Tanuj22) [reply package.json:79]: Okay

**Review by kushthedude — APPROVED** [review]

**Pull request merged**